packages/component-submission/server/use-cases/updateSubmissionUseCase.js

```diff
@@ -0,0 +1,60 @@
+const { pick } = require('lodash')
+
+function getManuscriptData(data) {
+  return pick(data, [
+    'meta',
+    'coverLetter',
+    'opposedSeniorEditorsReason',
+    'opposedReviewingEditorsReason',
+    'previouslyDiscussed',
+    'previouslySubmitted',
+    'cosubmission',
+    'opposedReviewersReason',
+    'submitterSignature',
+    'disclosureConsent',
+  ])
+}
+
+function getEditorTeams(data) {
+  return pick(data, [
+    'suggestedSeniorEditors',
+    'opposedSeniorEditors',
+    'suggestedReviewingEditors',
+    'opposedReviewingEditors',
+  ])
+}
+
+function getReviewerTeams(data) {
+  return pick(data, ['suggestedReviewers', 'opposedReviewers'])
+}
+
+const initialize = ({ submission, logger }) => ({
+  execute: async (manuscriptId, userId, data) => {
+    const manuscriptData = getManuscriptData(data)
+    const editorTeams = getEditorTeams(data)
+    const reviewerTeams = getReviewerTeams(data)
+
+    try {
+      await submission.initialize(manuscriptId, userId)
+
+      await submission.updateManuscript(manuscriptData)
+      await submission.updateAuthorTeam(data.author)
+      await submission.updateEditorTeams(editorTeams)
+      await submission.updateReviewerTeams(reviewerTeams)
+    } catch (error) {
+      const expected = 'Data Integrity Error'
+      if (error.message.startsWith(expected)) {
+        logger.error(`Expected a ${expected}, ${error.message}`)
```

Comment:
Is it safe to just swallow these errors - is there anything we need to do to clean up after the error occurs? Also which of the above statements throws - if it throws will it leave the stored data in an error state or will the transaction be aborted?

Reply:
Upon further inspection, if one of the `submission.update*` fails, the there's potential for a partial object to be stored and the error be swallowed resulting in :warning:  **loss of data**. :warning: 

I suggest a change is made to each of the `submission.update*` functions allowing the user to pass in a `transaction` object that's used by the internal database calls so these updates can be rolled back if one of them fails.

Reply:
The problem is that internally it calls the Manuscript model's save method which has a transaction and throws if the record has a later timestamp. Not sure how nesting transactions work and rewriting the manuscript save method is too big a change. I do agree however that it should be all one transaction

Reply:
Maybe we could just delete everything related to this `Manuscript` if an error occurs? The way it is currently, if an error happens at any stage we're going to end up with partial/corrupted data in the database, I can't think of anything else we could do without introducing transactions at this level.

Reply:
I think some of the teams not saved is acceptable, for now and would be reflected on the client once the new data has been fetched.